src/use-swr-pages.tsx

```diff
@@ -118,26 +119,39 @@ export function useSWRPages<OffsetType = any, Data = any, Error = any>(
 
     // if dataList is [], we can assume this page is empty
     // TODO: this API is not stable
-    if (dataList && !dataList.length) {
-      emptyPageRef.current = true
-    } else {
-      emptyPageRef.current = false
-    }
-
+    emptyPageRef.current = dataList && !dataList.length
     return dataList
   }, [])
 
+  const resetPages = useCallback(() => {
+    setPageSWRs([])
+    setPageCount(() => {
+      cache.set(pageCountKey, 1)
+      return 1
+    })
+    setPageOffsets(() => {
+      cache.set(pageOffsetKey, [null])
+      return [null]
+    })
+    pageCacheMap.set(pageKey, [])
+  }, [pageOffsetKey, pageCountKey])
+
+  // Reset page if the key changes
+  useEffect(() => resetPages(), [resetPages])
```

Comment:
Shouldn't `deps` be used as the dependencies for this effect?

As also done originally:
https://github.com/gabrielperales/swr/commit/8ba8379d2e6c580ded234385d34ea829be996788#diff-c9d2eb3ea821da85bd697b22fe31edf4R141

Reply:
deps itself can't be a dependency cause it is different everytime. You can't spread it either cause react will complain dependency array can't be different length during different render.

Reply:
I agree deps itself cannot be a dependency, but I also agree with ftes in saying that I could spread it alongside resetPages (I expect the user to provide a dependency array with the same size every render). Another issue is if we need to reset the pages every time a value used inside the Page component gets updated. In the original API, there is no deps array for the pages, so I use the pages key as a dependency. Maybe we need some clarification about the deps argument that from the code owner.